Enable MCP by default in helm values and e2e tests (#10222)

ayj opened the pull request:
(cherry picked from commit 970e0797c24d8995d7b6b0eba12d46c777dbb671)

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/10256#" title="Author self-approved">ayj</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **linsun**

If they are not already assigned, you can assign the PR to them by writing `/assign @linsun` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["linsun"]} -->

istio-testing commented:
@ayj: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | c56cd02ee11ccf4249119fab09657d81a4751a2d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/10256/istio-unit-tests/17441) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

costinm commented:
Also if I remember correctly - new targets need some magic setup to get the
grid working again.

Better to keep the names - history and less headaches.

On Mon, Dec 3, 2018 at 3:28 PM Jason Young <notifications@github.com> wrote:

> /test e2e-dashboard
> /test istio-unit-tests
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/10256#issuecomment-443910487>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAFI6qZhSPhAXq9PD6Q7SiTwB9za2DOUks5u1bO5gaJpZM4Y_lzm>
> .
>

ayj commented:
Agreed that we'll need to update testgrid. We already did that in the past when we added MCP test variants. We'll need to do the same for this PR.

@costinm, long term how would you suggest we make this change to avoid renames? 

We've changed default behavior in the past. How is this change any different? We'll know the cut-over date for comparison purposes.

costinm commented:
As long as you don't forget to update the testgrid - it's probably ok.
Since we had both tests on testgrid - keeping them with the original names
and grids
seemed cleaner, not sure why the renames were needed. We just need to make
the non-mcp not required.

On Mon, Dec 3, 2018 at 3:40 PM Jason Young <notifications@github.com> wrote:

> Agreed that we'll need to update testgrid. We already did that in the past
> when we added MCP test variants. We'll need to do the same for this PR.
>
> @costinm <https://github.com/costinm>, long term how would you suggest we
> make this change to avoid renames?
>
> We've changed default behavior in the past. How is this change any
> different? We'll know the cut-over date for comparison purposes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/10256#issuecomment-443913140>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAFI6iL2UtKtzBZq-ALEeNl6xGvjZlaMks5u1baIgaJpZM4Y_lzm>
> .
>

ayj commented:
Corresponding testgrid update is ready to be merged after this PR (see https://github.com/kubernetes/test-infra/pull/10334)

ozevren commented:
/lgtm

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/10256#" title="Author self-approved">ayj</a>*, *<a href="https://github.com/istio/istio/pull/10256#issuecomment-444273278" title="LGTM">ozevren</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **linsun**

If they are not already assigned, you can assign the PR to them by writing `/assign @linsun` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["linsun"]} -->
